contracts/token/ERC20/extensions/ERC7540.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {IERC7540} from "../../../interfaces/IERC7540.sol";
+import {IERC20} from "../../../interfaces/IERC20.sol";
+import {ERC4626} from "./ERC4626.sol";
+import {SafeERC20} from "../utils/SafeERC20.sol";
+import {Math} from "../../../utils/math/Math.sol";
+
+/**
+ * @dev Abstract implementation of the ERC-7540 standard, extending ERC-4626.
+ */
+abstract contract ERC7540 is ERC4626, IERC7540 {
+    using SafeERC20 for IERC20;
+    using Math for uint256;
+
+    // Mappings to track pending and claimable requests
+    mapping(address => mapping(uint256 => Request)) private _pendingDepositRequests;
+    mapping(address => mapping(uint256 => Request)) private _pendingRedeemRequests;
+
+    mapping(address => mapping(address => bool)) private _operators;
+
+    /**
+     * @dev Set the underlying asset contract.
+     */
+    constructor(IERC20 asset) ERC4626(asset) {}
+
+    /**
+     * @dev Creates a new deposit request.
+     */
+    function requestDeposit(
+        uint256 assets,
+        address controller,
+        address owner
+    ) external override returns (uint256 requestId) {
+        if (assets == 0) {
+            return 0;
+        }
```

Comment:
The ERC says:
> If a Vault returns 0 for the requestId of any request, it MUST return 0 for all requests.

Returning 0 here, and not for the other request is not compliant.

Reply:
Fixed